### PR TITLE
Fix `undefined` error due to `columnIdx` out of bounds 

### PR DESCRIPTION
**Pull request by amanmahajan7**
Closes https://github.com/adazzle/react-data-grid/issues/3369

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -690,6 +690,7 @@ function DataGrid<R, SR, K extends Key>(
   function isCellEditable(position: Position): boolean {
     return (
       isCellWithinViewportBounds(position) &&
+      position.idx !== -1 &&
```
Comment (nstepien) [inline src/DataGrid.tsx:693]: `isCellWithinViewportBounds` doesn't catch this? 🤔 
Should we add a test?

**Review by nstepien — APPROVED** [review]

File: src/DataGrid.tsx (hunk 1 above)
Comment (amanmahajan7) [reply src/DataGrid.tsx:693]: It uses https://github.com/adazzle/react-data-grid/blob/main/src/DataGrid.tsx#L674 and `minColIdx === -1`

Comment (amanmahajan7) [reply src/DataGrid.tsx:693]: May be we need to add a new function `withinEditBounds`

Comment (amanmahajan7) [reply src/DataGrid.tsx]: Not sure how to test it. I will check

**Review by nstepien — APPROVED** [review]

**Pull request merged**